### PR TITLE
Support "mjs" extensions

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: https://github.com/storybooks/storybook/issues/4125#issuecomment-423764765

## What I did 
Added mjs to the prod/dev webpack configs

### CODE REVIEW

File: lib/core/src/server/config/webpack.config.prod.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ export default ({ configDir, babelOptions, entries }) => {
     module: {
       rules: [
         {
-          test: /\.jsx?$/,
+          test: /\.(mjs|jsx?)$/,
```
[the remark this answers is not in the document]
Comment (libetl) [reply lib/core/src/server/config/webpack.config.prod.js]: (is it even possible ?)

[the remark this answers is not in the document]
Comment (igor-dv) [reply lib/core/src/server/config/webpack.config.prod.js]: yeah, I could set it to `m?jsx?`. But do they even exist?

**Review by libetl — APPROVED** [review]

File: lib/core/src/server/config/webpack.config.prod.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (libetl) [reply lib/core/src/server/config/webpack.config.prod.js]: good question

[the remark this answers is not in the document]
Comment (ndelangen) [reply lib/core/src/server/config/webpack.config.prod.js]: Let's wait until we hear from the community about this.

**Review by ndelangen — APPROVED** [review]

File: lib/core/src/server/config/webpack.config.prod.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hypnosphi) [reply lib/core/src/server/config/webpack.config.prod.js]: `m?jsx?` looks better to me

**Pull request merged**